src/subliminal/providers/opensubtitlescom.py

```diff
@@ -595,30 +595,27 @@ def api_get(
 
         return r.json()  # type: ignore[no-any-return]
 
-    def _search(self, *, page: int = 1, **params: Any) -> list[dict[str, Any]]:
-        # Extended params with page
-        ext_params = {'page': page, **params}
-
-        # query the server
-        logger.info('Searching subtitles %r', ext_params)
-
-        # GET request and add page information
-        response = self.api_get('subtitles', ext_params)
-
-        if not response or not response['data']:
-            return []
-
-        ret = response['data']
-
-        # check that the maximum number of pages has not been exceeded
-        allow_more_pages = self.max_result_pages <= 0 or page < self.max_result_pages
-
-        # retrieve other pages maybe
-        if allow_more_pages and 'total_pages' in response and page < response['total_pages']:
-            # missing pages
-            ret.extend(self._search(page=page + 1, **params))
+    def _search(self, **params: Any) -> Generator[dict[str, Any], None, None]:
+        page = 1
+        allow_more_pages = True
+        total_pages_reached = False
+        abort = False
+
+        while allow_more_pages and not total_pages_reached and not abort:
+            # Extended params with page
+            ext_params = {'page': page, **params}
+            logger.info('Searching subtitles %r', ext_params)
+            # GET request and add page information
+            response = self.api_get('subtitles', ext_params)
+            if not response or not response['data']:
+                abort = True
+            else:
+                yield from response['data']
 
-        return ret  # type: ignore[no-any-return]
+            page = page + 1
+            # check that the maximum number of pages has not been exceeded
+            allow_more_pages = self.max_result_pages <= 0 or page <= self.max_result_pages
```

Comment:
These two lines can be directly tested and `break` from the `while` loop if not passing.